src/igvfd/audit/biosample.py

```diff
@@ -40,5 +40,7 @@ def audit_biosample_taxa_check(value, system):
         if len(taxa_dict) > 1:
             detail = ''
             for k, v in taxa_dict.items():
-                detail += f'Biosample {audit_link(sample_id, sample_id)} has donors {audit_link(v, v)} that are {k}. '
+                donor_links = [audit_link(path_to_text(donor), donor) for donor in v]
+                donor_links = ', '.join(donor_links)
+                detail += f'Biosample {audit_link(path_to_text(sample_id), sample_id)} has donor(s) {donor_links} of taxa {k}. '
```

Comment:
i think it means the detail will say "Biosample A has donor B. Biosample A has donor G. Biosample A has donor T..." - i thin the biosample is redundant in such a detail?

Reply:
That's true, it's 1 sentence per taxa, so the Biosample appears for each donor taxa separately.
I assume it was written this way for organization/readability (to avoid a really long list of donors/taxas in one sentence). To avoid repeating Biosample, what do you think of this sentence construction: "Biosample {accession} has donors of taxas Homo sapiens {donorA, donorB}, Mus musculus {donor C, donor D}."  ?

Reply:
looks good to me